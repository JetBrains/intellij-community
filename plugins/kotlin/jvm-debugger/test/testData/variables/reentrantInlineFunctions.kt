package reentrantInlineFunctions

// SHOW_KOTLIN_VARIABLES

fun main() {
    val x = 0
    //Breakpoint!
    f(true) { x ->
        //Breakpoint!
        f(false) { x ->
            //Breakpoint!
            println()
        }
    }
}

inline fun f(b: Boolean, block: (Int) -> Unit) {
    if (b) {
        val x = 1
        //Breakpoint!
        block(x)
    } else {
        //Breakpoint!
        block(2)
    }
}

// IGNORE_K2