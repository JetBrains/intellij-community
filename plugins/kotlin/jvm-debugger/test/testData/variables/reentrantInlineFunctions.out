LineBreakpoint created at reentrantInlineFunctions.kt:8
LineBreakpoint created at reentrantInlineFunctions.kt:10
LineBreakpoint created at reentrantInlineFunctions.kt:12
LineBreakpoint created at reentrantInlineFunctions.kt:21
LineBreakpoint created at reentrantInlineFunctions.kt:24
Run Java
Connected to the target VM
reentrantInlineFunctions.kt:8
KotlinStackFrame (reentrantInlineFunctions.kt:8)
    JavaValue[local] x: int = 0 (reentrantInlineFunctions.kt:6)
reentrantInlineFunctions.kt:21
InlineStackFrame f (reentrantInlineFunctions.kt:21)
    JavaValue[local] b: boolean = true (reentrantInlineFunctions.kt:17)
    JavaValue[local] x: int = 1 (reentrantInlineFunctions.kt:19)
reentrantInlineFunctions.kt:10
InlineStackFrame lambda 'f' in 'main' (reentrantInlineFunctions.kt:10)
    JavaValue[local] x: int = 1 (reentrantInlineFunctions.kt:8)
reentrantInlineFunctions.kt:24
InlineStackFrame f (reentrantInlineFunctions.kt:24)
    JavaValue[local] b: boolean = false (reentrantInlineFunctions.kt:17)
reentrantInlineFunctions.kt:12
InlineStackFrame lambda 'f' in 'main' (reentrantInlineFunctions.kt:12)
    JavaValue[local] x: int = 2 (reentrantInlineFunctions.kt:10)
Disconnected from the target VM

Process finished with exit code 0
